### PR TITLE
Add GeoNetwork harvester based on the Q Search API endpoint.

### LINKED ISSUE #7: Explicitly handle duplicate dataset identifiers per source
It appears that some sources allocate duplicate dataset identifiers which we currently implicitly handle via last-write-wins. This should be replaced a explicit handling that decides which version should be used.

### PR DESCRIPTION
**Pull request by adamreichold**
This seems nicer than the CSW endpoint provided by GeoNetwork as it returns the full metadata immediately, but it does not seem to be significantly faster based on small tests.

Closes #7

### CODE REVIEW

**Comment by adamreichold** [comment]
This should also be used in the harvester configuration under version control.

**Comment by adamreichold** [comment]
A first test showed that improves thing significantly: Harvesting all of geodatenkatalog.de took a bit over 2 hours instead of almost 11.

**Comment by adamreichold** [comment]
Should make use of the `with_retry` helper introduced in #12 once that is merged.

File: src/harvester/geo_network_q.rs [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+    } else {
+        0
+    };
+
+    let results = response.metadata.len();
+    let mut errors = 0;
+
+    for entry in response.metadata {
+        if let Err(err) = translate_dataset(dir, source, entry).await {
+            tracing::error!("{:#}", err);
```
Comment (jakobdeller) [inline src/harvester/geo_network_q.rs:92]: I would not have expected translate_dataset to contain the write_dataset function. Maybe we find a more transparent naming?

Comment (jakobdeller) [reply src/harvester/geo_network_q.rs:92]: translate_write_dataset?

File: src/harvester/mod.rs [hunk 2]
```diff
@@ -1,19 +1,37 @@
...
 use serde::Deserialize;
 use tokio::time::{sleep, Duration};
 use toml::from_str;
 use url::Url;
 
+use crate::dataset::Dataset;
+
+async fn write_dataset(dir: &Dir, id: String, dataset: Dataset) -> Result<()> {
+    let file = match dir.open_with(&id, OpenOptions::new().write(true).create_new(true)) {
+        Ok(file) => file,
```
Comment (jakobdeller) [inline src/harvester/mod.rs:22]: Rename OpenOptions to FsOpenOptions for more clarity

**Review by jakobdeller — APPROVED** [review]

File: src/harvester/geo_network_q.rs (hunk 1 above)
Comment (adamreichold) [reply src/harvester/geo_network_q.rs:92]: I agree that the naming is unfortunate and will try to think of a better variant.

**Pull request merged**